### ISSUE TITLE
Introspection endpoint not implemented

### ISSUE THREAD

**Issue by michikuehne** [bshaffer/oauth2-server-php#606]
Hi there,

for our current project we are going to implement an introspection endpoint for our OpenID Connect server. This endpoint will will be used by the resource to validate the access token.

IEFT published a draft for the introspection endpoint December'14:
https://tools.ietf.org/html/draft-ietf-oauth-introspection-04

Are there any plans for implementing an introspection endpoint in this library as well?

Best,
Michael

**Comment by Raphy** [comment]
I'm wanting to implement an introspection endpoint for our `OAuth2 Authorization Server` to our `OAuth2 Resources Server` validate an `AccessToken`.

It could be nice to call something like : 
```php
$response = $server->handleIntrospectRequest($request, $response)
```

There is some news about the implementation of the introspection ?
The draft was validated into [RFC 7662 -  OAuth 2.0 Token Introspection](https://tools.ietf.org/html/rfc7662)

**Comment by UniBW** [comment]
Are there any news on the introspection endpoint?

Cheers
Andreas

**Comment by bshaffer** [comment]
I agree this should definitely be added! I'd love to get to it, but in the event this takes me a while to get to, pull requests are welcome!

**Comment by bdegoy** [comment]
Hello,
I have made a pull request from an "introspection" branch.
Regards,
Bertrand
